src/containers/BooksForm.js

```diff
@@ -0,0 +1,95 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+
+import { connect } from 'react-redux';
+import { createBook } from '../actions';
+
+export const categories = ['Action', 'Biography', 'History', 'Horror', 'Kids', 'Learning', 'Sci-Fi'];
+
+class BooksForm extends React.Component {
+  state = {
+    title: '',
+    category: '',
+  };
+
+  componentDidMount() {
```

Comment:
As mentioned in the previous review, we don't actually need `componentDidMount` for this purpose. In fact, we shouldn't be dispatching `createBook` when the component mounts.

The high level idea of what we should do:

1. We should have these books you are creating as initial state in `/reducers/book.js`:

```javascript
const books = (state = initialBooks, action) {
 // same as before
}
```

2. What this does is initialize your `books` state in the redux store with those dummy books. This is the right way to do this and not by dispatching `createBooks` everytime you mount the `BooksForm`

3. To display this initial `books` state is the job of `BookList` -> BookList is connecting to the `redux` store right? So `BookList` 's `mapStateToProps` return object should include a `books` attribute. This `books` attribute isn't simply `state.books` but a filtered list of the books:

```javascript
// BookList component

const mapStateToProps = state => ({
 filter: state.filter,
 books: state.books.filter(({category}) => state.filter === "" ? true : category === state.filter)
});
```
What the above means is that if the `state.filter` value is "" then simply return all the books in the redux store, otherwise return only the books whose category is equal to the value of state.filter.

Reply:
OK, got it now 👍 

As for filtering books, I implemented it a little bit differently, as ESLint was yelling at me. And in my opinion it is even better:
```javascript
const mapStateToProps = state => ({
  currentFilter: state.filter,
  books: state.filter
    ? state.books.filter(({ category }) => category === state.filter)
    : state.books,
});
```